library/src/main/java/org/apache/apex/malhar/lib/state/managed/BucketsFileSystem.java

```diff
@@ -139,13 +139,16 @@ protected void deleteBucket(long bucketId) throws IOException
    * @param data            data of all time-buckets
    * @throws IOException
    */
-  protected void writeBucketData(long windowId, long bucketId, Map<Slice, Bucket.BucketedValue> data) throws IOException
+  protected void writeBucketData(long windowId, long bucketId, Map<Slice, Bucket.BucketedValue> data, long latestPurgedTimeBucket) throws IOException
```

Comment:
This is used by writer service (async from operator) to write the data, but after it it also calls delete. The logic looks suspect and should at the very least be documented.